cloud/services/container/clusters/reconcile.go

```diff
@@ -320,25 +322,77 @@ func convertToSdkReleaseChannel(channel *infrav1exp.ReleaseChannel) containerpb.
 	}
 }
 
-func (s *Service) checkDiffAndPrepareUpdate(existingCluster *containerpb.Cluster) (bool, *containerpb.UpdateClusterRequest) {
+// convertToSdkMasterAuthorizedNetworksConfig converts the MasterAuthorizedNetworksConfig defined in CRs to the SDK version.
+func convertToSdkMasterAuthorizedNetworksConfig(config *infrav1exp.MasterAuthorizedNetworksConfig) *containerpb.MasterAuthorizedNetworksConfig {
+	if config == nil {
+		return nil
+	}
+
+	// Convert the CidrBlocks slice.
+	cidrBlocks := make([]*containerpb.MasterAuthorizedNetworksConfig_CidrBlock, len(config.CidrBlocks))
+	for i, cidrBlock := range config.CidrBlocks {
+		cidrBlocks[i] = &containerpb.MasterAuthorizedNetworksConfig_CidrBlock{
+			CidrBlock:   cidrBlock.CidrBlock,
+			DisplayName: cidrBlock.DisplayName,
+		}
+	}
+
+	return &containerpb.MasterAuthorizedNetworksConfig{
+		// Enabled specifies whether master authorized networks is enabled.
```

Comment:
nit: not sure you really need these comments when creating an instance of the struct.